Solutions/Marain.UserNotifications.Abstractions/Marain/UserNotifications/UserNotificationDeliveryStatus.cs

```diff
@@ -28,5 +28,10 @@ public enum UserNotificationDeliveryStatus
         /// The notification has been delivered.
         /// </summary>
         Delivered = 3,
+
+        /// <summary>
+        /// The notification has failed to be delivered.
+        /// </summary>
+        Failed = 4,
```

Comment:
So currently we have only intergrated with Airship and are only considering to log if the AirshipClient throws back an exception while sending a notifications. The next piece of work will also include the failure reason in a message field.

Reply:
What if airship just doesn't come back in time?

Reply:
i.e. a Timeout error on the calling side?

Reply:
Matthew's interpretation of what I mean is correct.

As an analogy, consider the difference between an HTTP response of 503 (Service Unavailable) vs 404 (Not found). If you get a 503, that tends to mean the service is temporarily having problems, and if you back off and retry, it might succeed next time. But if you get a 404, that indicates that the server was fully operational, and is confident that the thing you asked for doesn't exist. Retrying is unlikely to succeed in the case of a 404 unless you have good reason to believe that the underlying state of the system has changed.

Back to our scenario, "Airship had a bad moment there" is analogous to a 503, and a retry may well succeed. Conversely, if you receive a bounce saying `thisemailisbad@evil.org` is not a known mailbox, then that's essentially analogous to a 404, and retrying is not an appropriate strategy.

I can't tell which of these two styles of failure you're trying to represent here.

Reply:
Okay, we will add a retry policy for certain situations 503 vs 404 etc for different scenarios for Airship.

Reply:
Hi all, I have added the retry policy for a set of http status codes. if we still get back an exception, we save the error message as the failure reason in the delivery channel json for that saved notification.

Reply:
I think all we need to do to address this is rename the enum member to `PermanentlyFailed` (assuming that's what it still means). Assuming transient failures & retries are dealt with inside the delivery channel, notifications would have the status `Undelivered` while all that was taking place, and would only be updated to `PermanentlyFailed` when all the attempts at retrying had failed.

Reply:
Hi Jon, I have made the change for the enum to be `PermanentlyFailed` and pushed it just now.